tsl/src/dist_util.c

```diff
@@ -113,6 +127,33 @@ dist_util_set_id(Datum dist_id)
 					 (errmsg("database is already a member of a distributed database"))));
 	}
 
+	if (check_metadata)
+	{
+		uuid = ts_metadata_get_value(CStringGetDatum(METADATA_UUID_KEY_NAME),
+									 CSTRINGOID,
+									 UUIDOID,
+									 &isnull);
+		if (isnull)
+			ereport(ERROR,
+					(errcode(ERRCODE_TS_DATA_NODE_INVALID_CONFIG),
+					 (errmsg("uuid not set for data node"))));
+		else if (uuid_matches(uuid, dist_id))
+		{
+			Datum dist_uuid_str = DirectFunctionCall1(uuid_out, dist_id);
+
+			ereport(ERROR,
+					(errcode(ERRCODE_TS_DATA_NODE_SELF_REFERENCE),
+					 (errmsg("attempting to add self as data node"),
```

Comment:
This doesn't sound like an error to me, just a statement on what the user tried to do. I suggest:

```suggestion
					 (errmsg("cannot add the current database as a data node"),
```